FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeProcessingJobSpliter.cs

```diff
@@ -0,0 +1,319 @@
+using Microsoft.Health.AnalyticsConnector.DataClient.Api.Fhir;
+using Microsoft.Health.AnalyticsConnector.DataClient.Extensions;
+using Microsoft.Health.AnalyticsConnector.DataClient.Models.FhirApiOption;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Polly.Timeout;
+
+namespace Microsoft.Health.AnalyticsConnector.Core.Jobs
+{
+    public class FhirToDataLakeProcessingJobSpliter
+    {
+        private readonly IApiDataClient _dataClient;
+        private readonly IDiagnosticLogger _diagnosticLogger;
+        private readonly ILogger<FhirToDataLakeProcessingJobSpliter> _logger;
+        private const string CountApiParameter = "count";
+        private const string TotalCountFieldKey = "total";
+        private const string LastUpdatedApiParameter = "_lastUpdated";
+        private const string LastUpdatedApiParameterDesc = "-_lastUpdated";
+        private const int TimeoutSecondsGetResourceCount = 90;
+
+        public FhirToDataLakeProcessingJobSpliter(
+            IApiDataClient dataClient,
+            IDiagnosticLogger diagnosticLogger,
+            ILogger<FhirToDataLakeProcessingJobSpliter> logger)
+        {
+            _dataClient = EnsureArg.IsNotNull(dataClient, nameof(dataClient));
+            _diagnosticLogger = EnsureArg.IsNotNull(diagnosticLogger, nameof(diagnosticLogger));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public int LowBoundOfProcessingJobResourceCount { get; set; } = JobConfigurationConstants.LowBoundOfProcessingJobResourceCount;
+
+        public int HighBoundOfProcessingJobResourceCount { get; set; } = JobConfigurationConstants.HighBoundOfProcessingJobResourceCount;
+
+        public async IAsyncEnumerable<SubJobInfo> SplitJobAsync(string resourceType, DateTimeOffset? startTime, DateTimeOffset endTime, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            var totalCount = await GetResourceCountAsync(resourceType, startTime, endTime, cancellationToken);
+
+            if (totalCount <= HighBoundOfProcessingJobResourceCount)
+            {
+                // Return job with total count lower than high bound.
+                _logger.LogInformation($"Generate one {resourceType} job with {totalCount} count.");
+
+                yield return new SubJobInfo
+                {
+                    ResourceType = resourceType,
+                    TimeRange = new TimeRange() { DataStartTime = startTime, DataEndTime = endTime },
+                    ResourceCount = totalCount,
+                };
+            }
+            else
+            {
+                // Split large size job.
+                var splittingStartTime = DateTimeOffset.UtcNow;
+                _logger.LogInformation($"Start splitting {resourceType} job, total {totalCount} resources.");
+
+                var anchorList = await InitializeAnchorListAsync(resourceType, startTime, endTime, totalCount, cancellationToken);
```

Comment:
We'd better use the exact type instead of var here since it's hard to know the type of anchorList